src/JustSaying/JustSayingBus.cs

```diff
@@ -76,24 +89,23 @@ public void AddNotificationSubscriber(string region, INotificationSubscriber sub
                 // Just re-use existing subscriber instead.
                 return;
             }
-            subscribersForRegion[subscriber.Queue] = subscriber;
+            subscribersForRegion[queue.QueueName] = queue;
 
-            AddSubscribersToInterrogationResponse(subscriber);
+            // todo: this could work if we make the sqsqueue generic?
+            // otherwise can we do this in the AddMessageHandler bit instead?
+            // AddSubscribersToInterrogationResponse(queue);
         }
 
-        private void AddSubscribersToInterrogationResponse(INotificationSubscriberInterrogation interrogationSubscribers)
-        {
-            foreach (var subscriber in interrogationSubscribers.Subscribers)
-            {
-                _subscribers.Add(subscriber);
-            }
-        }
 
         public void AddMessageHandler<T>(string region, string queue, Func<IHandlerAsync<T>> futureHandler) where T : Message
         {
-            var subscribersByRegion = _subscribersByRegionAndQueue[region];
+            var handler = new MessageHandlerWrapper(MessageLock, Monitor, _loggerFactory);
+            var handlerFunc = handler.WrapMessageHandler(futureHandler);
+            HandlerMap.Add(typeof(T), handlerFunc);
+
+            /*var subscribersByRegion = _subscribersByRegionAndQueue[region];
```

Comment:
This one is perhaps worth a conversation. We've realised that there's perhaps no real need to track subscribers by region and queue as a) region failover support is possibly going away (?) and b) these handlers don't contribute to topology deployment, so it doesn't really matter if there's a single HandlerMap per bus instance. We've taken the approach of a single dispatcher/handler map that is stateless, and we use the message context to determine message type and look up the handler.

Can you see any problems with this approach?

Reply:
AFAIK we agreed to drop region support

Reply:
@martincostello do you have any info on region support? We don't currently support it in this PR, so I'm trying to decide where to leave the seam. If we're removing failover support, then it would seem to make sense to remove everything related to active regions and failover regions, as it would just become a single property on the config: `WithRegion(string region)`.

We could do this in a separate PR, but that would mean this PR would be broken in terms of multi-region support. Maybe that's ok!